### PR TITLE
IRIS-6093 | add Announcements extension for collecting active users

### PR DESCRIPTION
**Pull request by xkxd**
https://wikia-inc.atlassian.net/browse/IRIS-6093

## desc
we want to get active users on a specific wiki in a specific amount of time for Announcements (https://wikia-inc.atlassian.net/browse/IRIS-3915) purposes

@nikodamn

### CODE REVIEW

File: extensions/wikia/Announcements/models/Announcements.class.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	 *
+	 * @return mixed
+	 */
+	public function getActiveUsers( int $wikiId, int $period ) {
+		global $wgSpecialsDB;
+
+		$db = wfGetDB( DB_SLAVE, [], $wgSpecialsDB );
+		$userIds = [];
+
+		$query = $db->select(
```
Comment (mszabo-wikia) [inline extensions/wikia/Announcements/models/Announcements.class.php]: `selectFieldValues` helper allows you to immediately fetch values of user_id column into an array

File: extensions/wikia/Announcements/Announcements.setup.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+// Load needed classes
+$wgAutoloadClasses['AnnouncementsController'] = $dir . 'AnnouncementsController.class.php';
+
+// model classes
+$wgAutoloadClasses['Announcements'] = $dir . 'models/Announcements.class.php';
+
+// Add new controller to controllers list
+$wgWikiaApiControllers['AnnouncementsController'] = $dir . 'AnnouncementsController.class.php';
```
Comment (mszabo-wikia) [inline extensions/wikia/Announcements/Announcements.setup.php]: not needed

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/Announcements/AnnouncementsController.class.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	 *
+	 * @throws Error
+	 */
+	public function getActiveUsers() {
+
+		$wikiId = $this->request->getInt( self::WIKI_ID );
+		$period = $this->request->getInt( self::PERIOD );
+
+		if ( $period == 0 || $wikiId == 0 ) {
+			throw new Error( "You must define both wikiId and period" );
```
Comment (mszabo-wikia) [inline extensions/wikia/Announcements/AnnouncementsController.class.php]: let's throw `BadRequestException`, it will set proper status code 400

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/Announcements/AnnouncementsController.class.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+class AnnouncementsController extends WikiaController {
+
+	const WIKI_ID = 'wikiId';
+	const PERIOD = 'period';
```
Comment (rogatty) [inline extensions/wikia/Announcements/AnnouncementsController.class.php]: How about `days`?

**Review by rogatty — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

**Pull request merged**